app/utils/android-helpers.ts

```diff
@@ -0,0 +1,175 @@
+}
+
+export const configureAndroidNotifications = async () => {
+  notifee.createChannel({
+    id: "better-rail",
+    name: "Better Rail",
+    description: "Get live ride notifications",
+    importance: AndroidImportance.HIGH,
+    sound: "default",
+  })
+
+  notifee.createChannel({
+    id: "better-rail-live",
+    name: "Better Rail Live",
+    description: "Get live ride persistent notification",
+    vibration: false,
+  })
+
+  const onRecievedMessage = async (message: FirebaseMessagingTypes.RemoteMessage) => {
+    if (message.notification) {
+      Burnt.toast({
+        title: message.notification.title,
+        message: message.notification.body,
+      })
+    }
+
+    const state: RideState = message.data && {
+      status: message.data.status as RideStatus,
+      delay: Number(message.data.delay),
+      nextStationId: Number(message.data.nextStationId),
+    }
+
+    const rideNotificationId = await getRideNotificationId()
+    if (rideNotificationId && state) {
+      const rideRoute = await getRideRoute()
+      updateNotification(rideRoute, state)
+    }
+  }
+
+  messaging().onMessage(onRecievedMessage)
+  messaging().setBackgroundMessageHandler(onRecievedMessage)
+  notifee.onBackgroundEvent(() => {
+    return new Promise((resolve) => {
+      resolve()
+    })
+  })
+}
+
+export const startRideNotifications = async (route: RouteItem) => {
+  const token = await messaging().getToken()
+  const rideId = await rideApi.startRide(route, token)
+
+  if (!rideId) {
+    throw new Error("Couldn't start ride")
+  }
+
+  unsubscribeTokenUpdates = messaging().onTokenRefresh((newToken) => {
+    rideApi.updateRideToken(rideId, newToken)
+  })
+
+  await setRideRoute(route)
+  const nextStationId = findClosestStationInRoute(route)
+  const train = getTrainFromStationId(route, nextStationId)
+  const status = getRideStatus(route, train, nextStationId)
+
+  const state: RideState = {
+    status,
+    nextStationId,
+    delay: train.delay,
+  }
+
+  const rideNotificationId = await updateNotification(route, state)
+  await setRideNotificationId(rideNotificationId)
+  return rideId
+}
+
+export const cancelNotifications = async () => {
+  messaging().deleteToken()
+  if (unsubscribeTokenUpdates) unsubscribeTokenUpdates()
+
+  const rideNotificationId = await getRideNotificationId()
+  if (rideNotificationId) {
+    notifee.cancelNotification(rideNotificationId)
+    Preferences.clearMultiple(["rideRoute", "rideNotificationId"])
+  }
+}
+
+export const endRideNotifications = async (rideId: string) => {
+  await cancelNotifications()
+  return rideApi.endRide(rideId)
+}
+
+const updateNotification = async (route: RouteItem, state: RideState) => {
+  const rideNotificationId = await getRideNotificationId()
+  const userLanguage = (await Preferences.get("userLocale")) || getInitialLanguage()
+  i18n.locale = userLanguage
+
+  return notifee.displayNotification({
+    [rideNotificationId && "id"]: rideNotificationId,
```

Comment:
`&& "id"` will always resolve to true, isn't it?

Reply:
Not in the first run, it creates a random id and returns it. For some reason we can't just `id: rideNotificationId` even when `rideNotificationId  = undefined`.